src/components/feedback-particle-effect/feedback-text-effects.ts

```diff
@@ -1,159 +1,63 @@
-        );
-      } else {
-        this.context.fillText(
-          text,
-          this.textX,
-          this.textY + index * this.lineHeight
-        );
-      }
-    });
-    this.convertToParticle();
-  }
 
-  private handleTextWorkerMessage(event: MessageEvent): void {
-    this.particles = event.data.map(
-      ({ x, y, color }) => new TextParticle(this, x, y, color)
-    );
+    this.loadFont();
   }
 
-  private convertToParticle(): void {
-    const imageData = this.context.getImageData(
-      0,
-      0,
-      this.canvasWidth,
-      this.canvasHeight
-    );
-    this.context.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
-    this.textWorker.postMessage({
-      canvasWidth: this.canvasWidth,
-      canvasHeight: this.canvasHeight,
-      gap: this.gap,
-      pixels: imageData.data,
-    });
-  }
+  private async loadFont() {
+    const fontName = feedbackCustomFonts[lang] || feedbackTextDefault; // Determine the final font to use
+    const fontPath = `./assets/fonts/${fontName}.ttf`; // Construct the font path
 
-  public render(): void {
-    this.context.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
-    this.particles.forEach((particle) => {
-      particle.draw();
-      particle.update();
-    });
+    this.applyFontToElement(fontName, fontPath);
   }
 
-  public updateParticles(): void {
-    this.textWorker.postMessage({
-      particles: this.particles,
-      particleDuration: this.particleDuration,
-      startTime: this.startTime,
-    });
-  }
+  /**
+   * Applies the specified font to the feedback text element and adds the necessary @font-face rule.
+   * @param fontName - The name of the font to apply.
+   * @param fontPath - The path to the font file.
+   */
+  private applyFontToElement(fontName: string, fontPath: string) {
+    if (fontPath) {
+      const style = document.createElement("style");
+      style.textContent = `
+        @font-face {
+          font-family: '${fontName}';
+          src: url('${fontPath}') format('truetype');
+        }
+      `;
+      document.head.appendChild(style);
+    }
 
-  public clearParticle(): void {
-    this.particles = [];
+    if (this.feedbackTextElement) {
+      this.feedbackTextElement.style.fontFamily = `${fontName}, sans-serif`;
+    }
   }
 
-  public unregisterEventListener(): void {
-    this.textWorker.removeEventListener(
-      "message",
-      this.handleTextWorkerMessage
-    );
-    this.textWorker.terminate();
+  /**
+   * Updates the feedback text with an animation.
+   * @param text - The text to display in the feedback element.
+   */
+  public wrapText(text: string): void {
+    if (!this.feedbackTextElement) {
+      console.error("Feedback text element not found!");
+      return;
+    }
+
+    this.feedbackTextElement.textContent = text;
+    hideShowElement(false, this.feedbackTextElement);
+
+    setTimeout(() => hideShowElement(true, this.feedbackTextElement), 3000);
```

Comment:
@bernhardccodev made a changes here and keeps the constructor clean and separates initialization logic into its own method. please check

Reply:
@bernhardccodev i made an adjustment about memory leak prevention it will clears any existing setTimeout before creating a new one, preventing potential memory leaks.